Microsoft.NET.Build.Containers/CreateNewImage.cs

```diff
@@ -84,8 +93,56 @@ public CreateNewImage()
         Entrypoint = Array.Empty<ITaskItem>();
         EntrypointArgs = Array.Empty<ITaskItem>();
         Labels = Array.Empty<ITaskItem>();
+        ExposedPorts = Array.Empty<ITaskItem>();
     }
 
+    private void SetPorts(Image image, ITaskItem[] exposedPorts)
+    {
+        foreach (var port in exposedPorts)
+        {
+            var portNo = port.ItemSpec;
+            var portTy = port.GetMetadata("Type");
+            var parsePortResult = ContainerHelpers.ParsePort(portNo, portTy);
+            if (!parsePortResult.success)
+            {
+                ContainerHelpers.ParsePortError errors = (ContainerHelpers.ParsePortError)parsePortResult.parseErrors!;
+                var portString = portTy == null ? portNo : $"{portNo}/{portTy}";
+                if (errors.HasFlag(ContainerHelpers.ParsePortError.MissingPortNumber))
+                {
+                    Log.LogError("A ContainerPort item was provided without an Include metadata specifying the port number. Please provide a ContainerPort item with an ItemSpec: <ContainerPort Include=\"80\" />");
```

Comment:
Important for diagnosability IMO to provide the bogus input in the error:

```suggestion
                    Log.LogError("ContainerPort item '{0}' does not specify the port number. Please ensure the item's Include is a port number, for example '<ContainerPort Include=\"80\" />'", port.ItemSpec);
```